### PR TITLE
Remove CORCOMPILE_IMPORT_TABLE_ENTRY

### PR DESCRIPTION
**Pull request by A-And**
As per @jkotas 's suggestion, removing CORCOMPILE_IMPORT_TABLE_ENTRY as the wModuleRid field is not needed/supported in CoreCLR. 

[WIP] while I test nidump.exe

### CODE REVIEW

File: src/zap/zapimport.cpp [hunk 1]
```diff
@@ -69,7 +68,7 @@ ZapImportTable::ModuleReferenceEntry * ZapImportTable::GetModuleReference(CORINF
     pEntry->m_wAssemblyRid = (USHORT) assemblyIndex;
```
Comment (jkotas) [inline src/zap/zapimport.cpp]: `ZapImportTable` should be deleted too.  We should not need it at all.

Comment (A-And) [reply src/zap/zapimport.cpp]: If you're OK with these changes, I can merge them and submit the `ZapImportTable` refactor in a subsequent PR.

Comment (jkotas) [reply src/zap/zapimport.cpp]: I do not think the change is an improvement in the current state. It is really just renaming `CORCOMPILE_IMPORT_TABLE_ENTRY` to `USHORT`.

Comment (A-And) [reply src/zap/zapimport.cpp]: Reasonable. I'll close this and re-open once I have something more substantial.

**Pull request closed without merging**